src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/QueueIntegrationTest.kt

```diff
@@ -142,69 +144,36 @@ abstract class QueueIntegrationTest : IntegrationTest() {
-    setupIndexes()
-    val prisonerNumbers = prisoner.map { it.prisonerNumber }
-    assertThat(prisonerNumbers.groupingBy { it }.eachCount().filter { it.value != 1 }).hasSize(0)
-    prisonMockServer.stubFor(
-      WireMock.get(urlEqualTo("/api/offenders/ids"))
-        .willReturn(
-          WireMock.aResponse()
-            .withHeader("Content-Type", "application/json")
-            .withHeader("Total-Records", prisonerNumbers.size.toString())
-            .withBody(gson.toJson(prisonerNumbers.map { PrisonerSearchByPrisonerNumbersResourceTest.IDs(it) })),
-        ),
-    )
+    createIndexStatusIndex(SyncIndex.INDEX_B)
+    createPrisonerIndex(SyncIndex.INDEX_B)
+
     prisoner.forEach {
-      prisonMockServer.stubFor(
-        WireMock.get(urlEqualTo("/api/offenders/${it.prisonerNumber}"))
-          .willReturn(
-            WireMock.aResponse()
-              .withHeader("Content-Type", "application/json")
-              .withBody(it.toOffenderBooking()),
-          ),
-      )
-      incentivesMockServer.stubFor(
-        WireMock.get(urlPathEqualTo("/iep/reviews/booking/${it.bookingId}"))
-          .willReturn(
-            WireMock.aResponse()
-              .withHeader("Content-Type", "application/json")
-              .withBody(it.toIncentiveLevel()),
-          ),
-      )
+      val prisonerRequest = Request("PUT", "/${SyncIndex.INDEX_B.indexName}/_doc/${it.prisonerNumber}").apply {
+        setJsonEntity(gson.toJson(it.toPrisoner()))
+      }
+      elasticsearchClient.lowLevelClient.performRequest(prisonerRequest)
     }
 
-    webTestClient.put().uri("/prisoner-index/build-index")
-      .headers(setAuthorisation(roles = listOf("ROLE_PRISONER_INDEX")))
-      .exchange()
-      .expectStatus().isOk
-
-    await.atMost(Duration.ofSeconds(60)) untilCallTo { prisonRequestCountFor("/api/offenders/${prisonerNumbers.last()}") } matches { it == 1 }
-    await.atMost(Duration.ofSeconds(60)) untilCallTo { getNumberOfMessagesCurrentlyOnIndexQueue() } matches { it == 0 }
-
-    webTestClient.put().uri("/prisoner-index/mark-complete")
-      .headers(setAuthorisation(roles = listOf("ROLE_PRISONER_INDEX")))
-      .exchange()
-      .expectStatus().isOk
+    await untilCallTo { elasticsearchClient.count(CountRequest(SyncIndex.INDEX_B.indexName), RequestOptions.DEFAULT).count } matches { it == prisoner.size.toLong() }
   }
 
   internal fun createIndexStatusIndex(syncIndex: SyncIndex = SyncIndex.INDEX_A) {
-    val response = elasticsearchClient.lowLevelClient.performRequest(Request("HEAD", "/offender-index-status"))
-    if (response.statusLine.statusCode == 404) {
-      val indexOperations = elasticsearchOperations.indexOps(IndexCoordinates.of("offender-index-status"))
-      indexOperations.create()
-      indexOperations.putMapping(indexOperations.createMapping(IndexStatus::class.java))
+    elasticsearchClient.lowLevelClient.performRequest(Request("DELETE", "/offender-index-status"))
```

Comment:
Might be a good opportunity to tidy up the language and replace offender with prisoner here.